File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_Skillgroup.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Skillgroup : ITableOperate
{ private const string TAB_FILE_DATA = "skillgroup.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_SKILL_LEVEL1,
ID_SKILL_LEVEL2,
ID_SKILL_LEVEL3,
ID_SKILL_LEVEL4,
ID_SKILL_LEVEL5,
ID_SKILL_LEVEL6,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int[] m_SkillLevel = new int[6];
 public int GetSkillLevelbyIndex(int idx) {
 if(idx>=0 && idx<6) return m_SkillLevel[idx];
 return -1;
 }

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Skillgroup _values = new Tab_Skillgroup();
 _values.m_SkillLevel [ 0 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_LEVEL1] as string);
_values.m_SkillLevel [ 1 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_LEVEL2] as string);
_values.m_SkillLevel [ 2 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_LEVEL3] as string);
_values.m_SkillLevel [ 3 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_LEVEL4] as string);
_values.m_SkillLevel [ 4 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_LEVEL5] as string);
_values.m_SkillLevel [ 5 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_LEVEL6] as string);

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Skillgroup.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Skillgroup implements ITableOperate {
 public final static String TAB_FILE_DATA = "/skillgroup.txt";
 public static final int ID_SKILL_LEVEL1 = 0 ;
public static final int ID_SKILL_LEVEL2 = 1 ;
public static final int ID_SKILL_LEVEL3 = 2 ;
public static final int ID_SKILL_LEVEL4 = 3 ;
public static final int ID_SKILL_LEVEL5 = 4 ;
public static final int ID_SKILL_LEVEL6 = 5 ;
public static final int MAX_RECORD = 6 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_SkillLevel[] = new int[6];
 public int GetSkillLevelbyIndex(int idx) {
 if(idx>=0 && idx<6) return m_SkillLevel[idx];
 return -1;
 }

public boolean LoadTable(HashMap<Integer, Table_Skillgroup> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Skillgroup> _hash = (HashMap<Integer, Table_Skillgroup> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Skillgroup _values = new Table_Skillgroup();
 _values.m_SkillLevel [ 0 ] =  Integer.parseInt((String)valuesList.get(ID_SKILL_LEVEL1));
_values.m_SkillLevel [ 1 ] =  Integer.parseInt((String)valuesList.get(ID_SKILL_LEVEL2));
_values.m_SkillLevel [ 2 ] =  Integer.parseInt((String)valuesList.get(ID_SKILL_LEVEL3));
_values.m_SkillLevel [ 3 ] =  Integer.parseInt((String)valuesList.get(ID_SKILL_LEVEL4));
_values.m_SkillLevel [ 4 ] =  Integer.parseInt((String)valuesList.get(ID_SKILL_LEVEL5));
_values.m_SkillLevel [ 5 ] =  Integer.parseInt((String)valuesList.get(ID_SKILL_LEVEL6));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

